bridges/source/cpp_uno/gcc3_linux_loongarch64/cpp2uno.cxx

```diff
@@ -0,0 +1,474 @@
+extern "C" void vtableSlotCall(
+    unsigned long gpr0, unsigned long gpr1, unsigned long gpr2,
+    unsigned long gpr3, unsigned long gpr4, unsigned long gpr5,
+    unsigned long gpr6, unsigned long gpr7, double fpr0, double fpr1,
+    double fpr2, double fpr3, double fpr4, double fpr5, double fpr6,
+    double fpr7, ...)
+{
+    register void * volatile indirectRet asm ("$t0");
+    register sal_Int32 volatile functionIndex asm ("$t6");
+    register sal_Int32 volatile vtableOffset asm ("$t7");
+    va_list ap;
+    va_start(ap, fpr7);
+    assert(sizeof (va_list) == sizeof (loongarch64_va_list));
+    unsigned long gpr[8];
+    gpr[0] = gpr0;
+    gpr[1] = gpr1;
+    gpr[2] = gpr2;
+    gpr[3] = gpr3;
+    gpr[4] = gpr4;
+    gpr[5] = gpr5;
+    gpr[6] = gpr6;
+    gpr[7] = gpr7;
+    double fpr[8];
+    fpr[0] = fpr0;
+    fpr[1] = fpr1;
+    fpr[2] = fpr2;
+    fpr[3] = fpr3;
+    fpr[4] = fpr4;
+    fpr[5] = fpr5;
+    fpr[6] = fpr6;
+    fpr[7] = fpr7;
+    vtableCall(
+        functionIndex, vtableOffset, gpr,
+        reinterpret_cast<unsigned long *>(fpr),
+        static_cast<unsigned long *>(
+            reinterpret_cast<loongarch64_va_list *>(&ap)->stack),
+        indirectRet);
+    asm volatile(
+        "ld.d $r4, %[gpr_], 0\n\t"
+        "ld.d $r5, %[gpr_], 8\n\t"
+        "fld.d $f0, %[fpr_],0\n\t"
+        "fld.d $f1, %[fpr_],8\n\t"
+        "fld.d $f2, %[fpr_],16\n\t"
+        "fld.d $f3, %[fpr_],24\n\t"
+        :: [gpr_]"r" (gpr), [fpr_]"r" (fpr)
+        : "$r4", "$r5", "$f0", "$f1", "$f2", "$f3");
+    va_end(ap);
+}
+#pragma GCC diagnostic pop
+
+std::size_t const codeSnippetSize = 0x24;
+
+unsigned char * generateCodeSnippet(
+    unsigned char * code, sal_Int32 functionIndex, sal_Int32 vtableOffset)
+{
+    // lu12i.w $t6, <high functionIndex>
+    reinterpret_cast<unsigned int *>(code)[0] = 0x14000012
+        | (((functionIndex >> 12) & 0x000fffff) << 5);
+    // ori $t6, $t6, <low functionIndex>
+    reinterpret_cast<unsigned int *>(code)[1] = 0x03800252
+        | ((functionIndex & 0x00000fff) << 10);
+    // lu12i.w $t7, <low vtableOffset>
+    reinterpret_cast<unsigned int *>(code)[2] = 0x14000013
+        | (((vtableOffset >> 12) & 0x000fffff) << 5);
+    // ori $t7, $t7, <high vtableOffset> 
+    reinterpret_cast<unsigned int *>(code)[3] = 0x03800273
+        | ((vtableOffset & 0x00000fff) << 10);
+    // load vtableSlotCall to $t8
+    reinterpret_cast<unsigned int *>(code)[4] = 0x14000014
+    	| (((((unsigned long)vtableSlotCall) >> 12) & 0x000fffff) << 5);
+    reinterpret_cast<unsigned int *>(code)[5] = 0x03800294
+	| ((((unsigned long)vtableSlotCall) & 0x00000fff) << 10);
+    reinterpret_cast<unsigned int *>(code)[6] = 0x16000014
+	| (((((unsigned long)vtableSlotCall) >> 32) & 0x000fffff) << 5);
+    reinterpret_cast<unsigned int *>(code)[7] = 0x03000294
+	| (((((unsigned long)vtableSlotCall) >> 52) & 0x00000fff) << 10);
+    // jump $t8
+    reinterpret_cast<unsigned int *>(code)[8] = 0x4c000280;
```

Comment:
mention the assembly in comments for these insns, similar to the 4 insns above?

Reply:
I have revised it again